commit c35680f542ce291a13cd9c40d57e83c7630512f4
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Mon Feb 13 13:11:41 2017 +0000

    staging: bcm2835-camera: Correctly denote key frames in encoded data
    
    Forward MMAL key frame flags to the V4L2 buffers.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
index bb82f4b040c3..d8b43b7677e1 100644
--- a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
+++ b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
@@ -398,6 +398,9 @@ static void buffer_cb(struct vchiq_mmal_instance *instance,
 			dev->capture.last_timestamp = buf->vb.vb2_buf.timestamp;
 
 			vb2_set_plane_payload(&buf->vb.vb2_buf, 0, length);
+			if (mmal_flags & MMAL_BUFFER_HEADER_FLAG_KEYFRAME)
+				buf->vb.flags |= V4L2_BUF_FLAG_KEYFRAME;
+
 			v4l2_dbg(1, bcm2835_v4l2_debug, &dev->v4l2_dev,
 				 "Buffer has ts %llu",
 				 dev->capture.last_timestamp);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1711_c35680f542ce291a13cd9c40d57e83c7630512f4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1711_c35680f542ce291a13cd9c40d57e83c7630512f4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
